### ISSUE TITLE
[BUG] Lower case HTTP method resulting in 400

### ISSUE THREAD

**Issue by vdusek** [yifeikong/curl_cffi#360]
### Describe the bug

- Lower case HTTP method (e.g. `'get'`) results in 400 Bad Request.

### To Reproduce

```python
import asyncio
from curl_cffi.requests import AsyncSession

async def main():
    session = AsyncSession()
    response = await session.request('get', 'https://httpbin.org/get')
    print(f'status_code: {response.status_code}')
    print(f'content: {response.content[:1000].decode()}')

if __name__ == '__main__':
    asyncio.run(main())
```

Resulting in:

```text
status_code: 400
content: <html>
<head><title>400 Bad Request</title></head>
<body>
<center><h1>400 Bad Request</h1></center>
</body>
</html>
```

### Expected behavior

- It could work or at least let me know, that the method has to be uppercase.

### Versions

 - `curl_cffi` version: 0.7.1

**Comment by yifeikong** [comment]
Only upper case is supported, as specified in the [RFC](https://www.rfc-editor.org/rfc/rfc7231#section-4.1).

**Comment by vdusek** [comment]
> Only upper case is supported, as specified in the [RFC](https://www.rfc-editor.org/rfc/rfc7231#section-4.1).

Great, in that case, I'd expect some `ValueError` exception to be raised.

**Comment by yifeikong** [comment]
> > > Only upper case is supported, as specified in the [RFC](https://www.rfc-editor.org/rfc/rfc7231#section-4.1).
> > 
> > 
> > Great, in that case, I'd expect some `ValueError` exception to be raised.
> 
> Or this could be specified using type hints, such as `Literal["GET", "POST", "DELETE"]`

Good idea, fixed in b8f27474d7dfd30f3537d506b819586fa39d089f and 45ab712902d6724afc4a1c39ada24bf2c56206fc
